src/Business Foundation/App/NoSeries/src/Setup/NoSeriesSetupImpl.Codeunit.al

```diff
@@ -101,6 +101,7 @@ codeunit 305 "No. Series - Setup Impl."
         NoSeriesLine.SetCurrentKey("Series Code", "Starting Date");
         NoSeriesLine.SetRange("Series Code", NoSeriesRec.Code);
```

Comment:
I would extract these three lines into a function in the setup impl. codeunit like below. This allows anyone to set filters without potentially messing up with the record. Pure setting of filters which is what we need and they have access to what code and date we are trying to filter to.
GetNoSeriesLineFilters(var NoSeriesLine: Record "No. Series Line", code, date)
var
  var NoSeriesLine2: Record "No. Series Line"
begin
	NoSeriesLine2.SetRange("Starting Date", 0D, date);
	NoSeriesLine2.SetRange("Series Code", code)
	RaiseSetAdditionalNoSeriesLineFilters(NoSeriesLine2)
	If NoSeriesLine2.<filter for code field> <> code then
                Error(CodeFieldChangedErr); // Extensions should never change the code field range, this is a bug that developers should know immediately.
	NoSeriesLine.CopyFilters(NoSeriesLine2);
end;

Reply:
After extracting this, I think it looks pretty good :)

Reply:
I hope I understood you correctly about outsourcing the code, otherwise just let me know what the function should look like.